src/probnum/filtsmooth/gaussfiltsmooth/kalman_utils.py

```diff
@@ -0,0 +1,215 @@
+        time=time,
+        _linearise_at=_linearise_at,
+    )
+    crosscov = info["crosscov"]
+    updated_rv = condition_state_on_measurement(
+        pred_rv=rv, meas_rv=meas_rv, crosscov=crosscov, data=data
+    )
+    return updated_rv, meas_rv, {}
+
+
+def condition_state_on_measurement(pred_rv, meas_rv, crosscov, data):
+    """Condition a Gaussian random variable on an observation."""
+    new_mean = pred_rv.mean + crosscov @ np.linalg.solve(
+        meas_rv.cov, data - meas_rv.mean
+    )
+    new_cov = pred_rv.cov - crosscov @ np.linalg.solve(meas_rv.cov, crosscov.T)
+    updated_rv = pnrv.Normal(new_mean, new_cov)
+    return updated_rv
+
+
+def iterate_update(update_fun, stopcrit=None):
+    """Iterated update decorator."""
+    if stopcrit is None:
+        stopcrit = StoppingCriterion()
+
+    def new_update_fun(*args, **kwargs):
+        return _iterated_update(update_fun, stopcrit, *args, **kwargs)
+
+    return new_update_fun
+
+
+def _iterated_update(
+    update_fun, stopcrit, measurement_model, rv, time, data, _linearise_at=None
+):
+    """Turn an update_*() function into an iterated update.
+
+    This iteration is continued until it reaches a fixed-point (as measured with atol and rtol).
+    Using this inside `Kalman` yields the iterated (extended/unscented/...) Kalman filter.
+
+    Examples
+    --------
+    >>> import functools as ft
+    >>>
+    >>> stopcrit = StoppingCriterion(atol=1e-2, rtol=1e-4, maxit=1000)
+    >>> iterated_update_classic = ft.partial(iterated_update, update_fun=update_classic, stopcrit=stopcrit)
+    """
+    current_rv, meas_rv, info = update_fun(
+        measurement_model=measurement_model,
+        rv=rv,
+        time=time,
+        data=data,
+        _linearise_at=_linearise_at,
+    )
+    new_mean = current_rv.mean
+    old_mean = np.inf * np.ones(current_rv.mean.shape)
+    while not stopcrit.terminate(error=new_mean - old_mean, reference=new_mean):
+        old_mean = new_mean
+        current_rv, meas_rv, info = update_fun(
+            measurement_model=measurement_model,
+            rv=current_rv,
+            time=time,
+            data=data,
+        )
+        new_mean = current_rv.mean
+    return current_rv, meas_rv, info
+
+
+########################################################################################################################
+# Smoothing choices
+########################################################################################################################
+
+
+def rts_with_precon(smooth_step_fun):
+    """Make a smoothing step respect preconditioning.
+
+    Currently, this is only available for IBM() integrators.
+
+    Examples
+    --------
+    >>> import functools as ft
+    >>> rts_smooth_step_classic_with_precon = rts_with_precon(rts_smooth_step_classic)
+    """
+
+    def new_smoothing_function(*args, **kwargs):
+        return _rts_smooth_step_with_precon(smooth_step_fun, *args, **kwargs)
+
+    return new_smoothing_function
+
```

Comment:
preocnditioning is also just a decorator now, therefore it will be compatible with any smoothing update immeditaly (including square-root!)